File: Murat/day_4.py
class Solution:
    # https://leetcode.com/problems/same-tree/submissions/1183274479
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if (not p and not q): return True
        if (not p or not q): return False
        
        if p.val != q.val: return False
        r, l = True, True
        r = self.isSameTree(p.left,q.left)
        l = self.isSameTree(p.right,q.right)
        
        return r and l

File: Ozan/day_4.java
// Solution for 100. Same Tree
// link: https://leetcode.com/problems/same-tree/description/


/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p==null && q==null)
            return true;
        boolean comparison = (p!=null && q==null) || (p==null && q!=null) || (p.val!=q.val);
        if (comparison)
            return false;

        return isSameTree(p.left,q.left) && isSameTree(p.right,q.right);
    }
}